### PR TITLE
Bobby branch smile

### PR DESCRIPTION
**Pull request by BobbyShmurner**

### CODE REVIEW

File: scoreboard.js [hunk 1]
```diff
@@ -0,0 +1,91 @@
+var scoreboard_data = JSON.parse(localStorage.getItem("scoreboard"));
+
+if (scoreboard_data == null) {
+	scoreboard_data = {};
+}
+
+name_labels = [
+	"Aaron Millen",
```
Comment (dandabs) [inline scoreboard.js]: whats this indentation?

File: scoreboard.js [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+	"charlotte",
+	"connor",
+	"dan",
+	"dylan",
+	"james",
+	"matthew"
+]
+
+function loadScoreboard() {
+	console.log("Loading scoreboard");
```
Comment (DylanMcKenzie) [inline scoreboard.js]: Remove console.log

Comment (dandabs) [inline scoreboard.js]: ahem

File: scoreboard.js [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+
+	data_sorted = [];
+
+	for (var i = 0; i < name_labels.length; i++) {
+		var name = name_labels[i];
+		data_sorted.push([name, scoreboard_data[name]]);
+	}
+
+	data_sorted.sort(function (a, b) {
+		return a[1] - b[1]; // Sort by reverse score
```
Comment (dandabs) [inline scoreboard.js]: comments should be self explanatory

File: scoreboard.js [hunk 4]
```diff
@@ -0,0 +1,91 @@
...
+
+function addScoreboardEntry(name, scoreboard_list, scoreboard_data, placement) {
+	placement_suffix = "th";
+
+	if (placement == 1) {
+		placement_suffix = "st";
+	} else if (placement == 2) {
+		placement_suffix = "nd";
+	} else if (placement == 3) {
+		placement_suffix = "rd";
```
Comment (dandabs) [inline scoreboard.js]: twenty third or twenty thirth

File: scoreboard.js [hunk 5]
```diff
@@ -0,0 +1,91 @@
...
+	}
+
+	scoreboard_list.innerHTML += `<li class="scoreboard-item">
```
Comment (dandabs) [inline scoreboard.js]: injection?

File: scoreboard.js [hunk 6]
```diff
@@ -0,0 +1,91 @@
...
+		<p>${placement}${placement_suffix}. ${name}</p>
+		<input type="text" id="${name}-scoreboard" value=${scoreboard_data[name]} oninput="saveData('${name}')" onfocusout="loadScoreboard()">
+	</li>`;
+}
+
+function saveData(name) {
+	elemtent = document.getElementById(`${name}-scoreboard`);
+	scoreboard_data[name] = parseInt(elemtent.value);
+
+	localStorage.setItem("scoreboard", JSON.stringify(scoreboard_data));
```
Comment (dandabs) [inline scoreboard.js]: try catch?

File: styles.css [hunk 7]
```diff
@@ -68,3 +71,31 @@ text {
 .animate {
   animation: sizeAndSpin 5s forwards;
 }
+
+
```
Comment (dandabs) [inline styles.css]: two lines

**Review by dandabs — CHANGES_REQUESTED** [review]

File: scoreboard.js [hunk 8]
```diff
@@ -0,0 +1,90 @@
...
+  data_sorted.sort(function (a, b) {
+    return a[1] - b[1]; // Sort by reverse score
+  });
+
+  for (var i = 0; i < data_sorted.length; i++) {
+    addScoreboardEntry(data_sorted[i][0], list, scoreboard_data, i + 1);
+  }
+}
+
+function addScoreboardEntry(name, scoreboard_list, scoreboard_data, placement) {
```
Comment (dandabs) [inline scoreboard.js:58]: what is scopreboard list

File: scoreboard.js [hunk 9]
```diff
@@ -0,0 +1,90 @@
...
+  placement_suffix = "th";
+
+  if (placement == 1) {
+    placement_suffix = "st";
+  } else if (placement == 2) {
```
Comment (dandabs) [inline scoreboard.js:63]: magic number

File: scoreboard.js [hunk 10]
```diff
@@ -0,0 +1,90 @@
...
+
+  for (var i = 0; i < data_sorted.length; i++) {
+    addScoreboardEntry(data_sorted[i][0], list, scoreboard_data, i + 1);
+  }
+}
+
+function addScoreboardEntry(name, scoreboard_list, scoreboard_data, placement) {
+  placement_suffix = "th";
+
+  if (placement == 1) {
```
Comment (dandabs) [inline scoreboard.js:61]: magic number

File: scoreboard.js [hunk 11]
```diff
@@ -0,0 +1,90 @@
...
+    placement_suffix = "st";
+  } else if (placement == 2) {
+    placement_suffix = "nd";
+  } else if (placement == 3) {
```
Comment (dandabs) [inline scoreboard.js:65]: magic number

File: scoreboard.js (hunk 4 above)
Comment (BobbyShmurner) [reply scoreboard.js]: we have 8 fucking names

File: scoreboard.js [hunk 12]
```diff
@@ -0,0 +1,90 @@
...
+  scoreboard_data[name] = parseInt(elemtent.value);
+
+  localStorage.setItem("scoreboard", JSON.stringify(scoreboard_data));
+  playRandomAudio();
+}
+
+function playRandomAudio() {
+  random_name = names[Math.floor(Math.random() * names.length)];
+
+  var audio = new Audio(`resources/audio/${random_name}.mp3`);
```
Comment (dandabs) [inline scoreboard.js:86]: magic number

File: scoreboard.js [hunk 13]
```diff
@@ -0,0 +1,90 @@
...
+function loadScoreboard() {
+  var list = document.getElementById("scoreboard_list");
+
+  scoreboard_list.innerHTML = "";
+
+  for (var i = 0; i < name_labels.length; i++) {
+    var name = name_labels[i];
+
+    if (scoreboard_data[name] == undefined) {
+      scoreboard_data[name] = 0;
```
Comment (dandabs) [inline scoreboard.js:38]: magic number

File: styles.css [hunk 14]
```diff
@@ -23,17 +23,20 @@ text {
...
   width: 500px;
   height: 500px;
 }
+
 #question {
   width: 400px;
 }
+
 #question h1 {
   font-size: 50px;
```
Comment (dandabs) [inline styles.css:37]: not responsive should be rem or %

File: styles.css [hunk 15]
```diff
@@ -23,17 +23,20 @@ text {
...
   pointer-events: none;
 }
+
 #chart {
   width: 500px;
   height: 500px;
 }
+
 #question {
   width: 400px;
```
Comment (dandabs) [inline styles.css:33]: not responsive

File: styles.css [hunk 16]
```diff
@@ -23,17 +23,20 @@ text {
   font-size: 12px;
   pointer-events: none;
 }
+
 #chart {
   width: 500px;
```
Comment (dandabs) [inline styles.css:28]: not responsive

File: styles.css [hunk 17]
```diff
@@ -23,17 +23,20 @@ text {
   font-size: 12px;
```
Comment (dandabs) [inline styles.css:23]: not responsive should be rem or em

File: scoreboard.js (hunk 5 above)
Comment (BobbyShmurner) [reply scoreboard.js]: u can already inject html by right clicking and pressing "Inspect"

File: styles.css [hunk 18]
```diff
@@ -68,3 +71,31 @@ text {
 .animate {
   animation: sizeAndSpin 5s forwards;
```
Comment (dandabs) [inline styles.css:72]: magic number should be a css variable

File: scoreboard.js (hunk 6 above)
Comment (BobbyShmurner) [reply scoreboard.js]: fuck off

File: scoreboard.js [hunk 19]
```diff
@@ -0,0 +1,90 @@
+var scoreboard_data = JSON.parse(localStorage.getItem("scoreboard"));
+
+if (scoreboard_data == null) {
+  scoreboard_data = {};
+}
+
+name_labels = [
```
Comment (dandabs) [inline scoreboard.js:7]: should these two arrays be an object instead?

File: styles.css (hunk 7 above)
Comment (BobbyShmurner) [reply styles.css]: ok fair enough

File: scoreboard.js [hunk 20]
```diff
@@ -0,0 +1,90 @@
+var scoreboard_data = JSON.parse(localStorage.getItem("scoreboard"));
+
+if (scoreboard_data == null) {
```
Comment (dandabs) [inline scoreboard.js:3]: could you not do this in one line? like above json.parse || {}

**Review by dandabs — CHANGES_REQUESTED** [review]

File: scoreboard.js (hunk 12 above)
Comment (DylanMcKenzie) [reply scoreboard.js:86]: enum please

**Comment by BobbyShmurner** [comment]
fuck off im merging

**Comment by dandabs** [comment]
> fuck off im merging

matthew this is not appropriate for the workplace

File: scoreboard.js (hunk 8 above)
Comment (MatthewJKainos) [reply scoreboard.js:58]: i dont see a scopreboard list

**Pull request merged**